Ntuplizer/plugins/JpsiTauNtuplizer.cc

```diff
@@ -0,0 +1,1587 @@
+      nBranches_->JpsiTau_B_lips.push_back(cands[ic].cand_b_lips);
+      nBranches_->JpsiTau_B_pvip.push_back(cands[ic].cand_b_pvip);
+      nBranches_->JpsiTau_B_pvips.push_back(cands[ic].cand_b_pvips);
+      nBranches_->JpsiTau_B_fls3d.push_back(cands[ic].cand_b_fls3d);
+      nBranches_->JpsiTau_B_fl3d.push_back(cands[ic].cand_b_fl3d);
+      nBranches_->JpsiTau_B_alpha.push_back(cands[ic].cand_b_alpha);
+
+      std::vector<RefCountedKinematicParticle> allParticles4doc;
+      
+      allParticles4doc.push_back(pFactory.particle(tt1_muon, muon_mass, chi, ndf, muon_sigma));
+      allParticles4doc.push_back(pFactory.particle(tt2_muon, muon_mass, chi, ndf, muon_sigma));
+      allParticles4doc.push_back(pFactory.particle(mytracks[cands[ic].cand_tau_id1], pion_mass, chi, ndf, pion_sigma));
+      allParticles4doc.push_back(pFactory.particle(mytracks[cands[ic].cand_tau_id2], pion_mass, chi, ndf, pion_sigma));
+      allParticles4doc.push_back(pFactory.particle(mytracks[cands[ic].cand_tau_id3], pion_mass, chi, ndf, pion_sigma));
+
+
+      nBranches_->JpsiTau_B_maxdoca.push_back(getMaxDoca(allParticles4doc));
+      nBranches_->JpsiTau_B_mindoca.push_back(getMinDoca(allParticles4doc));
+      nBranches_->JpsiTau_B_vx.push_back(cands[ic].cand_b_vx);
+      nBranches_->JpsiTau_B_vy.push_back(cands[ic].cand_b_vy);
+      nBranches_->JpsiTau_B_vz.push_back(cands[ic].cand_b_vz);
+
+      nBranches_->JpsiTau_B_iso.push_back(cands[ic].cand_tau_iso);
+      nBranches_->JpsiTau_B_iso_ntracks.push_back(cands[ic].cand_tau_iso_ntracks);
+      nBranches_->JpsiTau_B_iso_mindoca.push_back(cands[ic].cand_tau_iso_mindoca);
+
+
+
+
+
+      /********************************************************************
+       *
+       * Step10: check gen-matching and fill them
+       *
+       ********************************************************************/
+
+  
+      TVector3 genvertex(-9.,-9.,-9.);
+  
+      std::vector<const reco::Candidate*> gen_nr_mu;
+      std::vector<const reco::Candidate*> gen_jpsi_mu;
+  
+
+      if(isMC){
+    
+	for( unsigned p=0; p < genParticles_->size(); ++p){
+
+	  //      std::cout << "gen: " << (*genParticles_)[p].pdgId() << " " << (*genParticles_)[p].status() << std::endl;
+
+	  // Bc daughters loop
+	  if(TMath::Abs((*genParticles_)[p].pdgId())==541 && (*genParticles_)[p].status()==2){
+
+	    // retrieve production vertex
+	    genvertex = getVertex((*genParticles_)[p]);
+
+	    for(int idd = 0; idd < (int)(*genParticles_)[p].numberOfDaughters(); idd++){
+	      Int_t dpid = (*genParticles_)[p].daughter(idd)->pdgId();
+	      //	  std::cout << "\t -> " <<  << " " << (*genParticles_)[p].daughter(idd)->status()<< std::endl;
+	      if(TMath::Abs(dpid)==13) gen_nr_mu.push_back((*genParticles_)[p].daughter(idd));
+	    }
+	  }
+
+	  // J/psi loop
+	  if(TMath::Abs((*genParticles_)[p].pdgId())==443 && 
+	     (*genParticles_)[p].status()==2 && 
+	     TMath::Abs((*genParticles_)[p].mother(0)->pdgId())==541){
```

Comment:
Yuta I'm not sure if this line could be a problem in the case the Jpsi propagates before decaying https://github.com/UZHCMS/EXOVVNtuplizerRunII/pull/242/files#diff-cbb5e52534dc13683b6965e47383aeb7R1338

Reply:
ok. Do you know How to be modified? That being said the matching efficiency of 93% seems to be already indicative that majority of cases it is ok ?